wp-includes/sqlite/class-wp-sqlite-pdo-user-defined-functions.php

```diff
@@ -493,6 +494,54 @@ public function regexp( $pattern, $field ) {
 		return preg_match( $pattern, $field );
 	}
 
+	/**
+	 * Method to emulate MySQL REGEXP_REPLACE() function.
+	 *
+	 * @param string|array $pattern     Regular expression to search for (or array of strings).
+	 * @param string|array $replacement The string or an array with strings to replace.
+	 * @param string|array $field       Haystack.
+	 *
+	 * @return Array if the field parameter is an array, or a string otherwise.
+	 */
+	public function regexp_replace( $field, $pattern, $replacement ) {
```

Comment:
I just checked and in MySQL REGEXP_REPLACE() also [takes three optional parameters](https://dev.mysql.com/doc/refman/8.4/en/regexp.html#function_regexp-replace):

> pos: The position in expr at which to start the search. If omitted, the default is 1.
> occurrence: Which occurrence of a match to replace. If omitted, the default is 0 (which means “replace all occurrences”).
> match_type: A string that specifies how to perform matching. The meaning is as described for [REGEXP_LIKE()](https://dev.mysql.com/doc/refman/8.4/en/regexp.html#function_regexp-like).

Would you be up for adding a support for those? If not, that's fine, but let's at add them to the function signature and fail with an informative warning (`REGEXP_REPLACE() don't support non-default values for $pos (fourth argument), .... if you need it then you can contribute here: ....`) if they're set to non-default values.

Reply:
> Would you be up for adding a support for those?

I will try to do that!